### PR TITLE
Allow CanvasPattern or CanvasGradient as stroke style

### LINKED ISSUE #5865: Cannot set pattern as stroke color
Cannot set pattern as stroke color. It works well for fill color, but doesn't for stroke color. It simple turns black...

JsFiddle:
https://jsfiddle.net/nt6ae1Ld/11/ (line 20 vs. line 23)

### PR DESCRIPTION
**Pull request by fredj**
fixes #5865

### CODE REVIEW

**Comment by fredj** [comment]
Rendering tests added, thanks for any review

File: src/ol/style/stroke.js [hunk 1]
```diff
@@ -223,9 +223,17 @@ ol.style.Stroke.prototype.setWidth = function(width) {
  */
 ol.style.Stroke.prototype.getChecksum = function() {
   if (this.checksum_ === undefined) {
-    this.checksum_ = 's' +
-        (this.color_ ?
-            ol.color.asString(this.color_) : '-') + ',' +
+    this.checksum_ = 's';
+    if (this.color_) {
+      if (typeof this.color_ === 'string') {
+        this.checksum_ += ol.color.asString(this.color_);
```
Comment (ahocevar) [inline src/ol/style/stroke.js]: `this.color_` is a string already, so no need to call `ol.color.asString()`.

Comment (fredj) [inline src/ol/style/stroke.js]: fixed, thanks

**Comment by tpesl** [comment]
How can I use this code? I have only ol.js (the latest version). Where can I find some instructions?

**Comment by ahocevar** [comment]
Looks good now. Please merge.

**Comment by fredj** [comment]
Thanks for the review @ahocevar

**Pull request merged**